airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -109,12 +109,12 @@ def read(
                         f"The requested stream {configured_stream.stream.name} was not found in the source."
                         f" Available streams: {stream_instances.keys()}"
                     )
-                stream_is_available, error = stream_instance.check_availability(logger, self)
-                if not stream_is_available:
-                    logger.warning(f"Skipped syncing stream '{stream_instance.name}' because it was unavailable. Error: {error}")
-                    continue
                 try:
                     timer.start_event(f"Syncing stream {configured_stream.stream.name}")
+                    stream_is_available, reason = stream_instance.check_availability(logger, self)
```

Comment:
moved check availability call under common try/except which handles errors during usual stream read,
moved log messages which indicate start of the stream sync before availability check in to make to understand which stream is the source of errors